test/lint/lint-include-guards.sh

```diff
@@ -7,15 +7,19 @@
 #
 # Check include guards.
 
+set -euo pipefail
+IFS=$'\n\t'
 export LC_ALL=C
+
 HEADER_ID_PREFIX="ZCASH_"
 HEADER_ID_PREFIX_UPSTREAM="BITCOIN_"
 HEADER_ID_SUFFIX="_H"
 
 REGEXP_EXCLUDE_FILES_WITH_PREFIX="src/(crypto/ctaes/|leveldb/|crc32c/|secp256k1/|tinyformat.h|univalue/)"
 
 EXIT_CODE=0
-for HEADER_FILE in $(git ls-files -- "*.h" | grep -vE "^${REGEXP_EXCLUDE_FILES_WITH_PREFIX}")
+HEADER_FILES=$(git ls-files -- "*.h" | grep -vE "^${REGEXP_EXCLUDE_FILES_WITH_PREFIX}")
+for HEADER_FILE in $HEADER_FILES
```

Comment:
I just learned about [`shopt -s inherit_errexit`](https://dougrichardson.us/notes/fail-fast-bash-scripting.html), which seems like it’ll let the original version of this code error properly. I think that’s a better change than to restructure the code to avoid the subshell error handling.